snitch.go

```diff
@@ -172,13 +172,25 @@ func New(cfg *Config) (*Snitch, error) {
 		return nil, err
 	}
 
+	errCh := make(chan error, 0)
+
 	// Start register
-	go s.register(director.NewFreeLooper(director.FOREVER, make(chan error, 1)))
+	go func() {
+		if err := s.register(director.NewFreeLooper(director.FOREVER, make(chan error, 1))); err != nil {
```

Comment:
`s.register()` could panic before - I updated it to return an err instead that we can catch and pass up to caller via chan.

Reply:
And honestly same thing should be done in s.heartbeat probably too at some point...